matlab/CMakeLists.txt

```diff
@@ -305,16 +305,17 @@ target_include_directories(arrow_matlab PRIVATE ${CPP_SOURCE_DIR})
 target_include_directories(arrow_matlab PRIVATE ${ARROW_INCLUDE_DIR})
 target_compile_definitions(arrow_matlab PRIVATE ARROW_MATLAB_EXPORTING)
 
-set(mexfcn_sources mex/mexfcn.cc)
-list(TRANSFORM mexfcn_sources PREPEND ${CPP_SOURCE_DIR}/arrow/matlab/)
+set(mexcall_sources mex/call.cc)
+list(TRANSFORM mexcall_sources PREPEND ${CPP_SOURCE_DIR}/arrow/matlab/)
 
-# Build mexfcn MEX binary.
+# Build call MEX binary.
 matlab_add_mex(R2018a
-               NAME mexfcn
-               SRC ${mexfcn_sources}
+               NAME mexcall
```

Comment:
Will the name of shared library here end up being mexcall.so or mexcall.dll?  Since mex files are shared libraries, and they need to be unique on some platforms (namely linux) to avoid symbol collisions.  mexcall is still fairly generic.  

I would consider something more obfuscated like mlarrowmexcall.so

Reply:
Thank you for your feedback, Nick! That's a great point that you have brought up: even though the MATLAB API for calling the MEX function will be unique, as it is within a package, the shared library that is loaded may have a conflict. I will look into a solution for this and submit it to this branch (marking the branch as [WIP] for now).

Since we want to keep the API as clean as possible, it may be effective to wrap the MEX function with a MATLAB function that calls the more obfuscated name. This will also provide an opportunity to validate function arguments before passing them to the MEX layer.